test/unified_format.py

```diff
@@ -1181,19 +1194,30 @@ def check_events(self, spec):
             events = event_spec["events"]
             # Valid types: 'command', 'cmap'
             event_type = event_spec.get("eventType", "command")
+            ignore_extra_events = event_spec.get("ignoreExtraEvents", False)
+            server_connection_id = event_spec.get("serverConnectionId")
+
             assert event_type in ("command", "cmap")
 
             listener = self.entity_map.get_listener_for_client(client_name)
             actual_events = listener.get_events(event_type)
             if len(events) == 0:
-                self.assertEqual(actual_events, [])
+                if not ignore_extra_events:
+                    self.assertEqual(actual_events, [])
```

Comment:
What do you think of clamping actual_events when ignoreExtraEvents is set?:
```
if ignore_extra_events:
    actual_events = actual_events[:len(events)]
```

That way we only need to check ignore_extra_events in one place.

Reply:
Done